lib/Analysis/ConstantFolding.cpp

```diff
@@ -1406,6 +1412,9 @@ static double getValueAsDouble(ConstantFP *Op) {
 static Constant *ConstantFoldScalarCall(StringRef Name, unsigned IntrinsicID,
                                         Type *Ty, ArrayRef<Constant *> Operands,
                                         const TargetLibraryInfo *TLI) {
+  if (Constant *C = hlsl::ConstantFoldScalarCall(Name, Ty, Operands)) // HLSL Change - Try hlsl constant folding first.
```

Comment:
I left the rest of the function intact because it also does constant folding for various llvm intrinsics (like `bswap`). I thought it would be useful to keep those around in case they are generated by any llvm optimizations.

So if we don't think we need them I can just ifdef out the whole thing. My preference is to not litter the function with ifdefs. What do you think?